engine/proofview.mli

```diff
@@ -425,6 +425,18 @@ module Unsafe : sig
   (** [tclGETGOALS] returns the list of goals under focus. *)
   val tclGETGOALS : Evar.t list tactic
 
+  (** [tclSETSHELF gls] sets goals [gls] as the current shelf. *)
+  val tclSETSHELF : Evar.t list -> unit tactic
+
+  (** [tclGETSHELF] returns the list of goals on the shelf. *)
+  val tclGETSHELF : Evar.t list tactic
```

Comment:
In the shelf/unshelve_goals PR #6943 (previously part of unifall-infra) I called this shelf and I don't think it's unsafe.

Reply:
Right. So, do you suggest I move it out of `Unsafe` and give it the name `shelf`? That looks indeed better to me. (Or shall you do it as part of #6943?)